plugins/IntranetMeasurable/Tracker/RequestProcessor.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\IntranetMeasurable\Tracker;
+
+use Piwik\Container\StaticContainer;
+use Piwik\Plugins\IntranetMeasurable\Type;
+use Piwik\Tracker\Cache;
+use Piwik\Tracker\Request;
+
+class RequestProcessor extends \Piwik\Tracker\RequestProcessor
+{
+    private $didEnableSetting = false;
+    private $settingName = 'ini.Tracker.trust_visitors_cookies';
+
+    public function manipulateRequest(Request $request)
+    {
+        $site = Cache::getCacheWebsiteAttributes($request->getIdSite());
+        $isIntranetSite = !empty($site['type']) && $site['type'] === Type::ID;
+
+        if ($isIntranetSite && !StaticContainer::get($this->settingName)) {
+            $this->setTrustCookiesSetting(1);
+            $this->didEnableSetting = true;
+        } elseif ($this->didEnableSetting) {
+            // we reset it in case of bulk tracking with different sites etc
+            $this->setTrustCookiesSetting(0);
```

Comment:
Was actually thinking about this line. If `[General] trust_visitors_cookies` is set to 1, then this will undo that setting, right?

Reply:
only if we enabled it before, which we wouldn't have in this case when it is already enabled in https://github.com/matomo-org/matomo/pull/13473/files#diff-74231b57ce12fbc00990b9d88e26426dR26